### PR TITLE
Fortigate ban unban features

### PR DESCRIPTION
**Pull request by guykeller**
## Status 
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/17701

## Description
Added several commands to FortiGate: ban-ip, unban-ip, get-banned-ips.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by ronykoz** [comment]
@kirbles19 could you please take a look at:

- [ ] https://github.com/demisto/content/blob/Fortigate-Ban-Unban-Features/Integrations/FortiGate/FortiGate.yml
- [ ] https://github.com/demisto/content/blob/Fortigate-Ban-Unban-Features/Integrations/FortiGate/CHANGELOG.md

File: Integrations/FortiGate/FortiGate.yml [hunk 1]
```diff
@@ -437,6 +437,40 @@ script:
...
+  - name: fortigate-ban-ip
+    arguments:
+    - name: ip_address
+      required: true
+      description: List of IP Addresses to ban, comma separated. IPv4 and IPv6 addresses
+        are allowed. For example, "1.1.1.1,6.7.8.9".
+      isArray: true
+    - name: expiry
+      description: Time until expiry in seconds. 0 for indefinite ban.
+    description: Immediately add one or more users to the banned list.
```
Comment (ronykoz) [inline Integrations/FortiGate/FortiGate.yml]:
```suggestion
    description: Immediately add one or more IPs to the banned list.
```

File: Integrations/FortiGate/FortiGate.py [hunk 2]
```diff
@@ -429,6 +470,86 @@ def create_firewall_service_request(service_name, tcp_range, udp_range):
...
+    response = http_request('POST', uri_suffix, data=json.dumps(payload))
+    return response
+
+
+def ban_ip_command():
+    ip_addresses_string = demisto.args()['ip_address']
+    ip_addresses_array = argToList(ip_addresses_string)
+    for ip_address in ip_addresses_array:
+        if not is_ip_valid(ip_address, True):
+            return_error('Error: invalid IP address sent as argument.')
```
Comment (ronykoz) [inline Integrations/FortiGate/FortiGate.py]: Wondering if we should send the error after processing all the IPs, this way you just get an error and stop the program

+

Add a blank line after

File: Integrations/FortiGate/FortiGate.py [hunk 3]
```diff
@@ -429,6 +470,86 @@ def create_firewall_service_request(service_name, tcp_range, udp_range):
...
+    time_to_expire = demisto.args().get('expiry')
+    if time_to_expire:
+        time_to_expire = convert_arg_to_int(time_to_expire, 'expiry')
+    else:
+        # The default time to expiration is 0, which means infinite time (It will remain banned).
+        time_to_expire = 0
```
Comment (ronykoz) [inline Integrations/FortiGate/FortiGate.py]: blank line after

File: Integrations/FortiGate/FortiGate.py [hunk 4]
```diff
@@ -429,6 +470,86 @@ def create_firewall_service_request(service_name, tcp_range, udp_range):
...
+
+    response = http_request('POST', uri_suffix, data=json.dumps(payload))
+    return response
+
+
+def ban_ip_command():
+    ip_addresses_string = demisto.args()['ip_address']
+    ip_addresses_array = argToList(ip_addresses_string)
+    for ip_address in ip_addresses_array:
+        if not is_ip_valid(ip_address, True):
```
Comment (ronykoz) [inline Integrations/FortiGate/FortiGate.py]: The True seems off without it 
```suggestion
        if not is_ip_valid(ip_address, accept_v6_ips=True):
```

File: Integrations/FortiGate/FortiGate.py [hunk 5]
```diff
@@ -429,6 +470,86 @@ def create_firewall_service_request(service_name, tcp_range, udp_range):
...
+        'ip_addresses': ip_addresses_array
+    }
+    response = http_request('POST', uri_suffix, data=json.dumps(payload))
+    return response
+
+
+def unban_ip_command():
+    ip_addresses_string = demisto.args()['ip_address']
+    ip_addresses_array = argToList(ip_addresses_string)
+    for ip_address in ip_addresses_array:
```
Comment (ronykoz) [inline Integrations/FortiGate/FortiGate.py]: same here

**Comment by kirbles19** [comment]
@richardbluestone after reviewing the 3 new commands in the .yml and the RN in the .md, please fix the HTML formatting for the existing docs: https://support.demisto.com/hc/en-us/articles/360012690573-Fortinet-FortiGate

File: Integrations/FortiGate/FortiGate.py (hunk 2 above)
Comment (guykeller) [reply Integrations/FortiGate/FortiGate.py]: Could you explain the first comment please? What do you feel could be improved here, and how?

As for the blank line - will fix

**Comment by richardbluestone** [comment]
Hi I've updated the .yml and .mkd files.

**Comment by richardbluestone** [comment]
@kirbles19 This is now done

**Pull request merged**